Build issues on Mac OS X 10.11 (El Capitan)

It's not clear how to build this on a Mac (it has worked in the past).  Maybe we could
add a blurb to the README.

This is a tracking issue - I'll fix it.

I am trying to build on El Capitan too. I have problem installing gcc4.8 from MacPort so I installed gcc49 instead to compile the rocket chip simulator. While building the tool chain I run into this error:

100  116M  100  116M    0     0   267k      0  0:07:27  0:07:27 --:--:--  247k
./contrib/download_prerequisites: line 32: wget: command not found
make: *** [src/gcc] Error 1

Does upgrade to gcc5 helps?

Try 
$ brew install wget

There are a couple of dependencies.  I think some are (all?) are mentioned in
the README.md but otherwise we should add them.

Tommy


> On Jan 2, 2016, at 01:24 , alansuphd <notifications@github.com> wrote:
> 
> I am trying to build on El Capitan too. I have problem installing gcc4.8 from MacPort so I installed gcc49 instead to compile the rocket chip simulator. While building the tool chain I run into this error:
> 
> 100 116M 100 116M 0 0 267k 0 0:07:27 0:07:27 --:--:-- 247k
> ./contrib/download_prerequisites: line 32: wget: command not found
> make: *** [src/gcc] Error 1
> 
> Does upgrade to gcc5 helps?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/riscv/riscv-tools/issues/15#issuecomment-168375076>.
>

Thanks kionix!

Out of interest I change using Apple's clang to compile risk-tools, the RISC-V toolchain, with
CC=cc CXX=c++ ./build.sh

I double check to make sure cc and c++ are indeed the Apple clang installed at /usr/bin and use -v to make sure they are indeed Apple LLVM version 7.0.2 (clang-700.1.81).

After riscv-fester has been built and into the riscv-isa-sim build, the compilation errors out saying:

Configuring project riscv-isa-sim
Building project riscv-isa-sim
clang: warning: treating 'c-header' input as 'c++-header' when in C++ mode, this behavior is deprecated
In file included from ../riscv/insn_template.h:3:
In file included from ../riscv/mmu.h:10:
../riscv/processor.h:119:15: error: implicit instantiation of undefined template
      'std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> >'
  std::string isa;
              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/iosfwd:193:33: note: 
      template is declared here
    class _LIBCPP_TYPE_VIS_ONLY basic_string;
                                ^
1 error generated.
make: *** [insn_template.h.gch] Error 1

Checking into the code and cannot find problems there. For a quick verification to identify if this a compiler problem, at the exact same source tree I change the compiler to gcc4.9 then the riscv-isa-sim build successfully. Since my interest is to study RISC-V so, I will not pursue further with the Apple clang unless someone kind enough to address mistakes I make in using Apple clang.

Alan

--- a/README.md
+++ b/README.md
@@ -28,9 +28,11 @@ Ubuntu packages needed:
 Note: This requires GCC >= 4.8 for C++11 support (including thread_local).
 To use a compiler different than the default (for example on OS X), use:
 
-	$ CC=gcc-4.8 CXX=g++-4.8 ./build.sh
+	$ CC=gcc-5 CXX=g++-5 ./build.sh
 
+One way to get a working gcc is to install [Homebrew](http://brew.sh) and issue
 
+	$ brew install gcc
 
 
 # <a name="newlibman"></a>The RISC-V GCC/Newlib Toolchain Installation Manual

